File: 卡码/代码随想录/Go/二叉树/01-二叉树遍历方式/LeetCode_0104.go
//104. 二叉树的最大深度
package main

import (
	"container/list"
	. "binaryTree/types"
)

func maxDepth(root *TreeNode) int {
	q:=list.New()
	var depth int
	if root!=nil {
		q.PushBack(root)
	}
	for q.Len()>0 {
		size:=q.Len()
		depth++
		for size>0 {
			size--
			node:=q.Remove(q.Front()).(*TreeNode)
			if node.Left!=nil {
				q.PushBack(node.Left)
			}
			if node.Right!=nil {
				q.PushBack(node.Right)
			}
		}
	}
	return depth
}

File: 卡码/代码随想录/Python/二叉树/01-二叉树遍历方式/LeetCode_0104.py
#104. 二叉树的最大深度
from typing import *
from ..TreeNode import *

class Solution:
  def maxDepth(self, root: Optional[TreeNode]) -> int:
    depth:int=0
    q:List[TreeNode]=[]
    if root!=None:
      q.append(root)
    while len(q)>0:
      size:int=len(q)
      depth+=1
      for i in range(size):
        node:TreeNode=q[0]
        q=q[1:]
        if node.left:
          q.append(node.left)
        if node.right:
          q.append(node.right)
    return depth
